core/src/main/kotlin/validity/PreGenerationChecker.kt

```diff
@@ -0,0 +1,12 @@
+package org.jetbrains.dokka.validity
+
+interface PreGenerationChecker : () -> PreGenerationCheckerOutput {
+
+    override fun invoke(): PreGenerationCheckerOutput
+}
+
+data class PreGenerationCheckerOutput(val result: Boolean, val messages: List<String>) {
+
+    operator fun plus(pair: Pair<Boolean, List<String>>) =
```

Comment:
Imho this should have a generic list type parameter and be extracted so it can be used somewhere else

Reply:
Or we can even consider using [Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/) which brings some cool error handling functions

Reply:
Yes, we can consider.
After considering, I come to conclusion, that `Result` would be great here IF  ONLY Kotlin would allow it to be return type for functions. So I am sticking to what is.

Reply:
And I don't expect it to be used anywhere else.